src/queryBuilder/FilterContents.js

```diff
@@ -0,0 +1,81 @@
+import * as F from 'futil-js'
+import _ from 'lodash/fp'
+import React from 'react'
+import { inject, observer } from 'mobx-react'
+import { ModalPicker, Modal, FilteredPicker, Dynamic } from '../layout/'
+import { fieldsToOptions } from '../FilterAdder'
+import { partial } from '../utils/mobx-react-utils'
+
+let FieldPicker = partial(
+  {
+    Modal,
+    Picker: FilteredPicker,
+  },
+  ModalPicker
+)
+
+let FilterContents = inject(_.defaults)(
+  observer(({ node, root, fields, types = {} }) => (
+    <div
+      style={{
+        // lineHeight: '30px',
+        // minHeight: '34px',
+        display: 'flex',
+        width: '100%',
+      }}
+    >
+      <FieldPicker
+        label={
+          node.field
+            ? _.get(`${node.field}.label`, fields) || F.autoLabel(node.field)
+            : 'Pick a Field'
+        }
+        options={fieldsToOptions(fields)}
+        // TODO: consider type options in case this isn't safe, e.g. a field/type change action
+        onChange={field => root.mutate(node.path, { field })}
+      />
+      {node.field && (
+        <div style={{ margin: '0 5px' }}>
+          <select
+            onChange={({ target: { value } }) => {
+              root.typeChange(node, value)
+            }}
+            value={node.type}
+          >
+            {_.map(
+              x => (
+                <option key={x.value} value={x.value} disabled={x.disabled}>
+                  {x.label}
+                </option>
+              ),
+              [
+                {
+                  value: null,
+                  label: 'Select Type',
+                  disabled: node.type,
+                },
+                ...F.autoLabelOptions(
+                  _.get(`${node.field}.typeOptions`, fields)
+                ),
+              ]
+            )}
+          </select>
+        </div>
+      )}
+      {node.type && (
+        <div
+          style={{
```

Comment:
This object can definitely be declared outside to avoid having to re-generate it every time. Probably in the `styles` file you have.

Reply:
Again, let's wait until we have confirmation that this will meaningfully impact performance and isn't optimizing microseconds like counting backwards on a for loop. I'd rather keep this as close to the usage as possible, especially since we're so early on in these components and aren't sure if these styles will stay as they are or be radically changed soon.